### DISCUSSION TITLE
peacock error

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#18362] [Q&A General] [answered]
Hi Moose expert and user,

I try to execute the input file in peacock. The following error appears. I have no idea what happened. Before I succeeded in implementing the same file in peacock and everything is fine. But I failed when performing execution again.
<img width="570" alt="Screen Shot 2021-07-16 at 2 25 41 PM" src="https://user-images.githubusercontent.com/74226455/125998954-f016b911-19d0-4fd6-939e-71785917a892.png">
Thank you for your help.

**Comment by GiudGiud** [comment, marked as answer]
Hello

This looks like a missing python package error. 
Looks like you have activated the moose environment so that's good. Could you please try updating moose and its dependencies?

So `conda update --all` then `git fetch upstream`, `git rebase upstream/devel` OR `git checkout upstream/devel` depending on whether you are developing or just using.

Guillaume

**Reply by xchengood** [reply]
@GiudGiud. Thanks. I try it according to your instruction. One message shows below when trying 'git fetch upstream'.
<img width="481" alt="Screen Shot 2021-07-16 at 3 35 14 PM" src="https://user-images.githubusercontent.com/74226455/126005643-e585ff9a-d915-41dd-acbc-df76e140509b.png">
Any advices? Thank you.

**Reply by GiudGiud** [reply]
The remote is probably called origin not upstream on your machine

**Reply by aeslaughter** [reply]
@GiudGiud is correct, you are just missing a python package. Be sure your conda enviroment has "moose-tools" installed, it will give you all the packages needed for Peacock. Running `conda list` will show if it is there or not.

**Reply by xchengood** [reply]
> The remote is probably called origin not upstream on your machine

@GiudGiud Thanks. After updating conda, when I try 'git checkout origin/devel', the message below was got. Any advice? Thank you.
<img width="612" alt="Screen Shot 2021-07-20 at 10 11 26 AM" src="https://user-images.githubusercontent.com/74226455/126349469-aca0c7ed-0683-42b4-925b-d9abd36c6474.png">

**Reply by xchengood** [reply]
> @GiudGiud is correct, you are just missing a python package. Be sure your conda enviroment has "moose-tools" installed, it will give you all the packages needed for Peacock. Running `conda list` will show if it is there or not.

Hi @aeslaughter Thank you. I use 'conda list' to confirm there is the moose-tools. But peacock is still not working and shows same error.
<img width="593" alt="Screen Shot 2021-07-20 at 10 00 24 AM" src="https://user-images.githubusercontent.com/74226455/126349972-bdef1ee3-6024-4e96-946d-2aee6d11b5da.png">

**Reply by GiudGiud** [reply]
Well updating moose might not be as critical.
If you want to do it anyway:
It looks like there are some merge conflicts. You ll have to either deal with those or just get rid of your changes (git stash) before checking out the latest moose

The ciritical thing here is that your moose-tools package is severely outdated. 
Please follow the instructions in this post to delete and re-install the moose environment: https://github.com/idaholab/moose/discussions/18378
pasted here:
```
conda deactivate
conda env remove -n moose
conda install mamba
mamba create -n moose moose-libmesh moose-tools
conda activate moose
```

**Reply by xchengood** [reply]
> Well updating moose might not be as critical.
> If you want to do it anyway:
> It looks like there are some merge conflicts. You ll have to either deal with those or just get rid of your changes (git stash) before checking out the latest moose
> 
> The ciritical thing here is that your moose-tools package is severely outdated.
> Please follow the instructions in this post to delete and re-install the moose environment: #18378
> pasted here:
> 
> ```
> conda deactivate
> conda env remove -n moose
> conda install mamba
> mamba create -n moose moose-libmesh moose-tools
> conda activate moose
> ```

@GiudGiud Thank you. It works well now.
